tests/test_sourmash.py

```diff
@@ -72,6 +72,77 @@ def test_sourmash_info_verbose():
     assert "loaded from path" in err
 
 
+def test_load_pathlist_from_file_does_not_exist():
+    from sourmash.sourmash_args import load_pathlist_from_file
+    with pytest.raises(ValueError) as e:
+        load_pathlist_from_file("")
+    assert "file '' does not exist" in str(e.value)
+
+
+@utils.in_tempdir
+def test_load_pathlist_from_file_empty(c):
+    from sourmash.sourmash_args import load_pathlist_from_file
+    file_list = c.output("file_list")
+    with open(file_list, "w") as fp:
+        fp.write("")
+    with pytest.raises(ValueError) as e:
+        load_pathlist_from_file(file_list)
+    assert "pathlist is empty" in str(e.value)
+
+
+@utils.in_tempdir
+def test_load_pathlist_from_file_badly_formatted(c):
+    from sourmash.sourmash_args import load_pathlist_from_file
+    file_list = c.output("file_list")
+
+    with open(file_list, "w") as fp:
+        fp.write("{'a':1}")
+
+    with pytest.raises(ValueError) as e:
+        load_pathlist_from_file(file_list)
+    assert "pathlist contains a badly formatted file" in str(e.value)
+    
+
+@utils.in_tempdir
+def test_load_pathlist_from_file_badly_formatted_2(c):
+    from sourmash.sourmash_args import load_pathlist_from_file
+    file_list = c.output("file_list")
+    
+    sig1 = utils.get_test_data('compare/genome-s10.fa.gz.sig')
+    newdir = c.output('newdir')
+    os.mkdir(newdir)
+    new_file = os.path.join(newdir, 'sig1')
+    shutil.copyfile(sig1, new_file)
+
+    with open(file_list, "w") as fp:
+        fp.write(new_file + "\n")
+        fp.write("{'a':1}")
+
+    with pytest.raises(ValueError) as e:
+        load_pathlist_from_file(file_list)
+    assert "pathlist contains a badly formatted file" in str(e.value)
+
+
+@utils.in_tempdir
+def test_load_pathlist_from_file_duplicate(c):
+    from sourmash.sourmash_args import load_pathlist_from_file
+    file_list = c.output("file_list")
+
+    sig1 = utils.get_test_data('compare/genome-s10.fa.gz.sig')
+    newdir = c.output('newdir')
+    os.mkdir(newdir)
+    new_file = os.path.join(newdir, 'sig1')
+    shutil.copyfile(sig1, new_file)
+
+    with open(file_list, "w") as fp:
+        fp.write(new_file + "\n")
```

Comment:
note that you don't need to copy `sig1` to `new_file`, you can put the path for `sig1` directly into the file_list file. pls change ;).

Reply:
Oh I see, I will change that.

Reply:
Should I remove all of the following lines then?
`newdir = c.output('newdir')
os.mkdir(newdir)
new_file = os.path.join(newdir, 'sig1')
shutil.copyfile(sig1, new_file)`

And then, in `fp.write()` on line 138, should I put `sig1` instead of `new_file`?

Reply:
please give it a try!